src/gurt/hash.c

```diff
@@ -1304,13 +1304,21 @@ d_hhash_link_delete(struct d_hhash *hhash, struct d_hlink *hlink)
 void
 d_hhash_link_getref(struct d_hhash *hhash, struct d_hlink *hlink)
 {
-	d_hash_rec_addref(&hhash->ch_htable, &hlink->hl_link.rl_link);
+	ch_rec_addref(&hhash->ch_htable, &hlink->hl_link.rl_link);
 }
 
 void
 d_hhash_link_putref(struct d_hhash *hhash, struct d_hlink *hlink)
 {
-	d_hash_rec_decref(&hhash->ch_htable, &hlink->hl_link.rl_link);
+	/**
+	 * handle hash table should not use D_HASH_FT_EPHEMERAL,
+	 * just in case someone add it.
+	 */
```

Comment:
If this code is unreachable then better to add an assert when the hash table is created IMO.